doc/lsst.cp.pipe/constructing-calibrations.rst

```diff
@@ -0,0 +1,733 @@
+- Followed by running the defect pipeline:
+.. code:: bash
+
+    RERUN=20210706h
+    pipetask --long-log run -b /repo/main -p $CP_PIPE_DIR/pipelines/Latiss/findDefects.yaml \
+        -i LATISS/raw/all,u/czw/DM-28920/biasGen.20210702a,LATISS/calib \
+        -o u/czw/DM-28920/defectGen.$RERUN \
+        -d "instrument='LATISS' AND detector=0 AND exposure IN ($EXPOSURES)"  >& ./defect.$RERUN.log
+
+..
+
+  - For this test, certification was delayed until the entire chain of calibrations had been generated and verified.  This illustrates the fact that the butler can access calibrations from the RUN collection that they were generated in, that no other types of that calibration are found in a collection that is searched earlier.
+- Verification of the defects:
+.. code:: bash
+
+    pipetask --long-log run -b /repo/main -p $CP_VERIFY_DIR/pipelines/verifyDefect.yaml \
+        -i LATISS/raw/all,u/czw/DM-28920/defectGen.$RERUN,u/czw/DM-28920/biasGen.20210702a,LATISS/calib \
+        -o u/czw/DM-28920/verifyDefect.$RERUN \
+        -d "instrument='LATISS' AND detector=0 AND exposure IN ($EXPOSURES)" >& ./defectVerify.$RERUN.log
+
+..
+
+  - By placing the ``u/czw/DM-28920/defectGen.20210706h`` collection before the ``LATISS/calib`` collection, we can use the defects just created, and not the ingested defects that mask the entirety of amplifier ``C07``.
+  - As before, there will be a ``$CP_VERIFY_DIR/examples/cpVerifyDefects.ipynb`` containing the visualization and test failure information.
+  - It is also possible to rerun the bias verification, and confirm that these new defects improve the tests success.  That was the case here, with all failures on ``C04`` being resolved as well as some of the failures on ``C11``:
+.. code:: bash
+
+    pipetask --long-log run -b /repo/main -p $CP_VERIFY_DIR/pipelines/verifyBias.yaml \
+        -i LATISS/raw/all,u/czw/DM-28920/defectGen.20210702e,u/czw/DM-28920/biasGen.20210702a,LATISS/calib \
+        -o u/czw/DM-28920/verifyBias.$RERUN \
+        -d "instrument='LATISS' AND detector=0 AND exposure IN ($EXPOSURES)" \
+        -c verifyBiasApply:doDefect=True >& ./biasVerify.$RERUN.log
+
+- If the validation tests pass, the new defects can be certified:
+.. code:: bash
+
+    butler certify-calibrations /repo/main u/czw/DM-28920/defectGen.20210706h LATISS/calib/DM-28920 \
+         --begin-date 2020-01-01 --end-date 2050-01-01 defects
+
+
+.. _cp-pipe-darks:
+
+Constructing darks
+==================
+
+- As with biases, first identify the inputs:
+.. code:: bash
+
+  butler query-dimension-records /repo/main exposure \
+      --where "instrument='LATISS' AND exposure.observation_type='dark' \
+              AND exposure.exposure_time > 0.0 AND exposure.dark_time > 0.0 \
```

Comment:
I don't think we are actually writing dark_time properly in the files at the moment but I thought when it wasn't there that it defaulted to the exposure_time.

Reply:
The dark time constraint is there because I know there's a warning message in the assembly code that I didn't want to trigger.  Looking through the list of all darks, it doesn't look like there are any negative exp_time entries (which wasn't the case with biases).